crates/nu-command/src/filters/length.rs

```diff
@@ -15,27 +18,95 @@ impl Command for Length {
     }
 
     fn signature(&self) -> nu_protocol::Signature {
-        Signature::build("length").category(Category::Filters)
+        Signature::build("length")
+            .switch("column", "Show the number of columns in a table", Some('c'))
+            .category(Category::Filters)
     }
 
     fn run(
         &self,
-        _engine_state: &EngineState,
+        engine_state: &EngineState,
         _stack: &mut Stack,
         call: &Call,
         input: PipelineData,
-    ) -> Result<nu_protocol::PipelineData, nu_protocol::ShellError> {
-        match input {
-            PipelineData::Value(Value::Nothing { .. }, ..) => Ok(Value::Int {
-                val: 0,
-                span: call.head,
-            }
-            .into_pipeline_data()),
-            _ => Ok(Value::Int {
-                val: input.into_iter().count() as i64,
-                span: call.head,
-            }
-            .into_pipeline_data()),
+    ) -> Result<nu_protocol::PipelineData, ShellError> {
+        let col = call.has_flag("column");
+        if col {
+            length_col(engine_state, call, input)
+        } else {
+            length_row(call, input)
         }
     }
 }
+
+// this simulates calling input | columns | length
+fn length_col(
+    engine_state: &EngineState,
+    call: &Call,
+    input: PipelineData,
+) -> Result<PipelineData, ShellError> {
+    length_row(
+        call,
+        getcol(engine_state, call.head, input)
+            .expect("getcol() should not fail used in column command"),
+    )
+}
+
+fn length_row(call: &Call, input: PipelineData) -> Result<PipelineData, ShellError> {
+    match input {
+        PipelineData::Value(Value::Nothing { .. }, ..) => Ok(Value::Int {
+            val: 0,
+            span: call.head,
+        }
+        .into_pipeline_data()),
+        _ => Ok(Value::Int {
+            val: input.into_iter().count() as i64,
+            span: call.head,
+        }
+        .into_pipeline_data()),
+    }
+}
+
+fn getcol(
```

Comment:
You might want to look into `get_columns` which is currently in the table viewer. The reason I mention it is that the way we draw tables now, it's possible that a column missing in the first row might get added by a later row.

We could probably move `get_columns` to somewhere a bit more central so multiple commands could take advantage of it, too.

Reply:
yes, agreed, let me review the best way to move forward.  thanks for your feedback, I was not happy with the current implementation...